### ISSUE TITLE
Add the learning rate (in exponential representation, like "9.197948717948718e-08") to the model training output

### ISSUE THREAD

**Issue by blademoon** [huggingface/transformers#27631]
### Feature request

Good day. Can you add current learning rate to output of training process. 
Now, if we train model with Trainer we get those output (without leraning rate):

<img width="302" alt="image" src="https://github.com/huggingface/transformers/assets/17560478/239c3ce1-c7dc-4d79-9664-52ac4e3ea605">

I'm using this code to add information about the current learning rate value to the output:

```python
from transformers import Trainer, Seq2SeqTrainer
from typing import Any, Dict, List, Union

class MyTrainer(Seq2SeqTrainer):
    def log(self, logs: Dict[str, float]) -> None:
        logs["learning_rate"] = self._get_learning_rate()
        print(logs["learning_rate"])
        super().log(logs)
        

trainer = MyTrainer(
    args=training_args,
    model=model,
    train_dataset=common_voice["train"],
    eval_dataset=common_voice["test"],
    data_collator=data_collator,
    compute_metrics=compute_metrics,
    tokenizer=processor.feature_extractor,
)

trainer.train()
```

This causes the learning rate values to appear in the tabular output during model training, as here:

<img width="536" alt="image" src="https://github.com/huggingface/transformers/assets/17560478/defa1a31-15a5-4994-905e-7afdccee76de">

But the latest version of the library doesn't have enough field width to display this parameter correctly.

### Motivation

Currently, when using the library, the learning rate is not displayed. It can only be viewed on the hub. But if the library is used locally without internet access, there is no way to track the current learning rate during fine-tuning/training of the model.

### Your contribution

Gave a succinct example of the sample code in "Feature request" field that I use to output a learning rate value.

**Comment by ArthurZucker** [comment]
Hey! I think the way you log it is nice, just needs the log to be in scientific format!

**Comment by blademoon** [comment]
@ArthurZucker I already try convert `float` to scientific notation by using:
```python
...
    logs["learning_rate"] = "{:.2e}".format(self._get_learning_rate())
...
```

It dosent work. Tensorboard din't support this.

**Comment by ArthurZucker** [comment]
feel free to open a PR and ping @muellerzr !

**Comment by blademoon** [comment]
@ArthurZucker Maybe you know someone on the HuggingFace team who could help solve the "Rate" field width issue in the output?

**Comment by muellerzr** [comment]
@blademoon can you try using `pip install git+https://github.com/huggingface/transformers@muellerzr-use-scientific`? I think that should fix it :)

**Comment by blademoon** [comment]
@muellerzr Testing in progress...)))

<img width="605" alt="image" src="https://github.com/huggingface/transformers/assets/17560478/478bb935-2474-4c91-96a6-5ae470fdf396">

**Comment by blademoon** [comment]
@muellerzr 

Tested it 3 times.
First time - just installed the library.
Second time - restarted the kernel in notepad. Repeated the test.
Third time - restarted WSL2, to be more sure that the problem is reproduced after a reboot.

The result in all three cases:

<img width="560" alt="image" src="https://github.com/huggingface/transformers/assets/17560478/356d13d1-69f3-4b91-8415-96b5fbc1db9e">


The code used in the tests*:
```python
training_args = Seq2SeqTrainingArguments(
    output_dir = result_model_name,                       # измените имя репозитория по своему усмотрению
    per_device_train_batch_size=train_batch_size,
    gradient_accumulation_steps=accumulation_steps,       # увеличивается в 2x раза при каждом уменьшении размера батча в 2x раза  
    learning_rate=lr,
    lr_scheduler_type=lr_scheduler,                           # Планировщик для learning_rate
    warmup_steps=250,
    max_steps=training_steps_count,
    gradient_checkpointing=True,
    fp16=True,
    evaluation_strategy="steps",
    optim = "adamw_torch",
    per_device_eval_batch_size=eval_batch_size,        # Самый простой способ задать равным параметру per_device_train_batch_size
    predict_with_generate=True,
    generation_max_length=225,
    save_steps=500,
    eval_steps=500,
    logging_steps=25,
    report_to=["tensorboard"],
    load_best_model_at_end=True,
    metric_for_best_model=selected_metric,
    greater_is_better=False,
    push_to_hub=True,
    hub_private_repo=True
)


# class MyTrainer(Seq2SeqTrainer):
#     def log(self, logs: Dict[str, float]) -> None:
#         logs["learning_rate"] = self._get_learning_rate() * 1_000_000  # Умножение необходимо для исправления отображения. 
#         print(logs["learning_rate"])
#         super().log(logs)



# trainer = MyTrainer(
#     args=training_args,
#     model=model,
#     train_dataset=common_voice["train"],
#     eval_dataset=common_voice["test"],
#     data_collator=data_collator,
#     compute_metrics=compute_metrics,
#     tokenizer=processor.feature_extractor,
# )


trainer = Seq2SeqTrainer(
    args=training_args,
    model=model,
    train_dataset=common_voice["train"],
    eval_dataset=common_voice["test"],
    data_collator=data_collator,
    compute_metrics=compute_metrics,
    tokenizer=processor.feature_extractor,
)
```
(*) - Naturally, the commented lines were not executed.

**Comment by blademoon** [comment]
@muellerzr The learning rate is duplicated in scientific notation and decimalization for some unknown reason.

**Comment by muellerzr** [comment]
I’m still actively working on it, but your excitement is appreciated :)

**Comment by blademoon** [comment]
@muellerzr If additional testing is required. Tag me. Thanks again for your responsiveness.

**Comment by blademoon** [comment]
@ArthurZucker @muellerzr Good afternoon. I think you might be interested to know. Thanks to the feature you are developing, I was able to fine-tune Whisper small to WER = 12.6%.

https://huggingface.co/artyomboyko/whisper-small-fine_tuned-ru-v2
https://huggingface.co/spaces/artyomboyko/Whisper-medium-ru-v2

**Comment by blademoon** [comment]
@muellerzr Good day. Will this enhancement be implemented?

**Comment by qgallouedec** [comment]
Hey @artyomboyko, sorry for the very long silence here, we're catching up on the backlog.

Good news on the substance: the learning rate is already added to the training logs on every logging step, so it's fully trackable offline via the console/tqdm output, TensorBoard/W&B scalars, and `trainer.state.log_history`. The console printer even formats it in scientific notation (the `.4g` format), which was your original concern.

The one place it still doesn't appear is the Jupyter notebook progress table: `NotebookProgressCallback` uses a fixed column set https://github.com/huggingface/transformers/blob/150eb7c9ed4091294c829fa0e9466b090cb0f87f/src/transformers/utils/notebook.py#L312 and never renders an LR column. @muellerzr started #27660 for exactly this but it stalled. If you'd still like the column in the notebook table, a small PR adding `learning_rate` to those columns would be very welcome, and +1s here help us gauge demand. Leaving this open as a tracking thread in the meantime.
